### PR TITLE
Battle regressions fix

### PR DESCRIPTION
**Pull request by IvanSavenko**
Fixes regressions from my previous PR:
- #1249 - Status bar & in-game console may work incorrectly in some cases
- Transparency test for detecting currently hovered hex in battle is broken

### CODE REVIEW

File: client/battle/BattleInterfaceClasses.h [hunk 1]
```diff
@@ -55,12 +55,13 @@ class BattleConsole : public CIntObject, public IStatusBar
 	/// current text entered via in-game console, takes priority over both log entries and hover text
 	std::string consoleText;
 
-	/// if true then we are currently entering console text
+	/// if true then we are currently entering console tex
```
Comment (Nordsoft91) [inline client/battle/BattleInterfaceClasses.h:58]: needs rollback

File: client/widgets/TextControls.cpp [hunk 2]
```diff
@@ -426,16 +428,19 @@ void CGStatusBar::init()
 
 void CGStatusBar::clickLeft(tribool down, bool previousState)
 {
-	if(!down && onClick)
+	if(!down && onClick )
```
Comment (Nordsoft91) [inline client/widgets/TextControls.cpp:431]: excessive change

File: client/widgets/TextControls.cpp [hunk 3]
```diff
@@ -426,16 +428,19 @@ void CGStatusBar::init()
...
 	{
-		onClick();
+		if(LOCPLINT && LOCPLINT->cingconsole->active)
```
Comment (Nordsoft91) [inline client/widgets/TextControls.cpp:433]:
```suggestion
		if(LOCPLINT && LOCPLINT->cingconsole && LOCPINT->cingconsole->active)
```

**Review by Nordsoft91 — APPROVED** [review]

File: client/widgets/TextControls.cpp (hunk 3 above)
Comment (IvanSavenko) [reply client/widgets/TextControls.cpp:433]: Pointless - LOCPLINT->cingconsole is created in costructor and deleted in destructor. Rest of the code does not checks for cingconsole either

**Pull request merged**